### ISSUE TITLE
Syntax highlighting of automake files (Makefile.am)

### ISSUE THREAD

**Issue by Anteru** [pygments/pygments-migration-test#702]
(Original issue 995 created by fluzz on 2014-05-21T13:45:39.331161+00:00)

The "if" keyword is used in Makefile.am files to conditionally create Makefiles (see http://www.gnu.org/software/automake/manual/automake.html#Conditionals)

Currently, the "if" keyword is not recognized by the MakefileLexer, and thus an "err" token is generated (see http://pygments.org/demo/386955/)

So, please, could you add "if" in the keywords list of MakefileLexer ?

Thx for your great work !

**Comment by Anteru** [comment]
(Original issue was assigned to birkenfeld)

(Original issue 995 last updated on 2014-10-08T23:47:35.473592+00:00)

(Issue automaticaly closed due to status in Bitbucket: resolved)

**Comment by Anteru** [comment]
(Original comment by lev_lozhkin on 2014-05-21T17:36:20.369270+00:00)

Here's a quick patch that will give the desired behavior. I went for clarity, although that made part of the regex redundant.

```
#!diff

diff -r 2ba9b53c87ee pygments/lexers/text.py
--- a/pygments/lexers/text.py   Tue May 20 07:44:45 2014 -0700
+++ b/pygments/lexers/text.py   Wed May 21 10:26:38 2014 -0700
@@ -189,7 +189,9 @@
         # BSD Make
         r'\.\s*(include|undef|error|warning|if|else|elif|endif|for|endfor)|'
         # GNU Make
-        r'\s*(ifeq|ifneq|ifdef|ifndef|else|endif|-?include|define|endef|:))(?=\s)')
+        r'\s*(ifeq|ifneq|ifdef|ifndef|else|endif|-?include|define|endef|:)|'
+        # GNU Automake
+        r'\s*(if|else|endif))(?=\s)')
     r_comment = re.compile(r'^\s*@?#')

     def get_tokens_unprocessed(self, text):

```

**Comment by Anteru** [comment]
(Original comment by fluzz on 2014-05-21T17:51:21.366285+00:00)

I locally "hack-fixed" it by adding 'if' in GNU Make part of the regexp, just because I was wondering if redundancy would not have a bad impact on efficiency (I'm not at all a python expert, though).

May I expect your patch to be commited for next release ?

Thx.

**Comment by Anteru** [comment]
(Original comment by lev_lozhkin on 2014-05-21T18:28:09.980395+00:00)

I haven't contributed to this project before, so I am unfamiliar with its conventions. I will look into submitting a pull request within a day or two.

**Comment by Anteru** [comment]
(Original comment by tshatch on 2014-10-08T23:47:35.467706+00:00)

MakefileLexer: Add GNU Automake directives.

Fixes #995

→ <<cset 4cc0c2a0a420>>
